sql/server/src/main/java/io/stargate/api/sql/server/avatica/StargateMeta.java

```diff
@@ -320,8 +321,15 @@ public ExecuteBatchResult executeBatch(
   }
 
   @Override
-  public Frame fetch(StatementHandle h, long offset, int fetchMaxRowCount) {
-    throw new UnsupportedOperationException();
+  public Frame fetch(StatementHandle h, long offset, int fetchMaxRowCount)
+      throws NoSuchStatementException, MissingResultsException {
+    StatementHolder statement = statement(h);
+    ExecutionResult result = statement.result();
+    if (result == null) {
+      throw new MissingResultsException(h);
```

Comment:
Will `h` provide enough information or should we add some context to this exception?

Reply:
`MissingResultsException` is defined in Avatica, we have to provide `h`.

I'm not sure about an extended message... the context should be clear, I hope. This is kind of an internal failure mode where the client requests data without first executing something.